extensions/amp-story-shopping/0.1/amp-story-shopping-config.js

```diff
@@ -1,19 +1,199 @@
+  /* Optional Attrs */
+  'aggregateRating': [
+    createValidateConfigObject(aggregateRatingValidationConfig),
+  ],
+  'productIcon': [validateURLs],
+  'productTagText': [validateString],
+};
+
 /**
  * @typedef {{
  *  items: !Array<!ShoppingConfigDataDef>,
  * }}
  */
 let ShoppingConfigResponseDef;
 
+/**
+ * Validates an Object using the validateConfig function.
+ * @param {?Object=} validation
+ * @return {boolean}
+ */
+function createValidateConfigObject(validation) {
+  return (field, value) => {
+    if (!validateConfig(value, validation, field)) {
+      throw Error(
+        `Value for field '${field}' is not vaild, see the error messages above for details`
+      );
+    }
+  };
+}
+
+/**
+ * Validates an Array of Objects using the validateConfig function.
+ * @param {?Object=} validation
+ * @return {boolean}
+ */
+function createValidateConfigArray(validation) {
+  return (field, value) => {
+    let isValid = true;
+    for (const item of value) {
+      isValid &&= validateConfig(item, validation, field);
+    }
+    if (!isValid) {
+      `Value for field '${field}' is not vaild, see the error messages above for details`;
+    }
+  };
+}
+
+/**
+ * Validates if a required field exists for shopping config attributes
+ * @param {string} field
+ * @param {?string=} value
+ */
+export function validateRequired(field, value) {
+  if (value === undefined) {
+    throw Error(`Field ${field} is required.`);
+  }
+}
+
+/**
+ * Validates if string type for shopping config attributes
+ * @param {string} field
+ * @param {?string=} str
+ */
+export function validateString(field, str) {
+  if (typeof str !== 'string') {
+    throw Error(`${field} ${str} is not a string`);
+  }
+}
+
+/**
+ * Validates if it is a valid HTML ID for shopping config attributes
+ * @param {string} field
+ * @param {?string=} id
+ */
+export function validateHTMLId(field, id) {
+  const checkID = /^[A-Za-z]+[\w\-\:\.]*$/;
+  if (!checkID.test(id)) {
+    throw Error(`${field} ${id} is not a valid HTML Id`);
+  }
+}
+
+/**
+ * Validates number in shopping config attributes
+ * @param {string} field
+ * @param {?number=} number
+ */
+export function validateNumber(field, number) {
+  if (
+    (typeof number === 'string' && !/^[0-9.,]+$/.test(number)) ||
+    (typeof number !== 'string' && typeof number !== 'number')
+  ) {
+    throw Error(`Value ${number} for field ${field} is not a number`);
+  }
+}
+
+/**
+ * Validates currency in shopping config attributes
+ * @param {string} field
+ * @param {?string=} curr
+ */
+export function validateCurrency(field, curr) {
+  // This will throw an error on invalid currency codes.
+  Intl.NumberFormat('en-EN', {currency: curr}).format(0);
+}
+
+/**
+ * Validates url of shopping config attributes
+ * @param {string} field
+ * @param {?Array<string>=} url
+ */
+export function validateURLs(field, url) {
+  if (url === undefined) {
+    return;
+  }
+
+  const urls = Array.isArray(url) ? url : [url];
+
```

Comment:
Mind deleting this blank line and the one above it? I think that blank lines can be good for organizational purposes, but this function is short enough that they're less necessary